### PR TITLE
MIDI import changed into drumstick

### PR DESCRIPTION
**Pull request by senseab**
OK，这个功能我算是交差了
自己测试的结果还是不错的
你之前说 ds 有个与 Qt4 兼容的问题，这就交给你吧
如果代码里有不妥的地方，你就看着改好了
希望能尽快并入官方主线

### CODE REVIEW

**Comment by senseab** [comment]
干脆把导出也做了吧

**Comment by senseab** [comment]
导出功能坑太多了，先合并导入功能吧

**Comment by senseab** [comment]
我把 drumstick 支持的文件类型都做上了，感觉可以安心过年了

**Comment by senseab** [comment]
代码精简了

**Comment by liushuyu** [comment]
Rebased and hide your "placeholder" comments.
The next time, you can try `git stash` or `git commit --amend` to prevent from the commits whose commit message somewhat imply your contributions are non-sense or randomly written. This is certainly not what you want.

**Comment by senseab** [comment]
我好像把你的提交覆盖了，能再合并一次吗？
实在抱歉，我在调试一些 bug

File: plugins/SmfImport/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ INCLUDE(BuildPlugin)
 LINK_DIRECTORIES(${DRUMSTICK_FILE_LIBDIR})
 INCLUDE_DIRECTORIES(${DRUMSTICK_FILE_INCLUDEDIR})
 
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions -ldrumstick-file -Wno-reorder")
```
Comment (liushuyu) [inline plugins/SmfImport/CMakeLists.txt]: IMO, this is guaranteed not gonna work under Windows.

Comment (senseab) [reply plugins/SmfImport/CMakeLists.txt]: I have reverted to your commit and just fix that cannot link to drumstick-file problem, just see my new commit

**Pull request closed without merging**